lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -0,0 +1,1242 @@
+      OAILOG_ERROR_UE(
+          LOG_AMF_APP, imsi,
+          "Error could not update this ue context %p "
+          "gnb_ue_ngap_ue_id " GNB_UE_NGAP_ID_FMT
+          "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT " %s\n",
+          ue_context_p, ue_context_p->gnb_ue_ngap_id,
+          ue_context_p->amf_ue_ngap_id, hashtable_rc_code2string(h_rc));
+    }
+    ue_context_p->gnb_ngap_id_key = gnb_ngap_id_key;
+  }
+
+  if (amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID) {
+    if (ue_context_p->amf_ue_ngap_id != amf_ue_ngap_id) {
+      h_rc = hashtable_ts_remove(
+          amf_state_ue_id_ht, (const hash_key_t) ue_context_p->amf_ue_ngap_id,
+          (void**) &ue_context_p);
+      h_rc = hashtable_ts_insert(
+          amf_state_ue_id_ht, (const hash_key_t) amf_ue_ngap_id,
+          (void*) ue_context_p);
+
+      if (HASH_TABLE_OK != h_rc) {
+        // TODO: this method is deprecated and will be removed once the AMF's
+        // context is migrated to map in the upcoming multi-UE PR
+        OAILOG_ERROR(
+            LOG_AMF_APP,
+            "Insertion of Hash entry failed for  "
+            "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT PRIX32 " \n",
+            amf_ue_ngap_id);
+      }
+      ue_context_p->amf_ue_ngap_id = amf_ue_ngap_id;
+    }
+  } else {
+    // TODO: this method is deprecated and will be removed once the AMF's
+    // context is migrated to map in the upcoming multi-UE PR
+    OAILOG_ERROR(
+        LOG_AMF_APP, "Invalid  amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT PRIX32 " \n",
+        amf_ue_ngap_id);
+  }
+
+  h_rc = hashtable_uint64_ts_remove(
+      amf_ue_context_p->imsi_amf_ue_id_htbl,
+      (const hash_key_t) ue_context_p->amf_context.imsi64);
+
+  if (INVALID_AMF_UE_NGAP_ID != amf_ue_ngap_id) {
+    h_rc = hashtable_uint64_ts_insert(
+        amf_ue_context_p->imsi_amf_ue_id_htbl, (const hash_key_t) imsi,
+        amf_ue_ngap_id);
+  } else {
+    h_rc = HASH_TABLE_KEY_NOT_EXISTS;
+  }
+
+  if (HASH_TABLE_OK != h_rc) {
+    // TODO: this method is deprecated and will be removed once the AMF's
+    // context is migrated to map in the upcoming multi-UE PR
+    OAILOG_ERROR(
+        LOG_AMF_APP,
+        "Insertion of Hash entry failed for  "
+        "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT PRIX32 " \n",
+        amf_ue_ngap_id);
+  }
+
+  amf_directoryd_report_location(
+      ue_context_p->amf_context.imsi64, ue_context_p->amf_context.imsi.length);
+  h_rc = hashtable_uint64_ts_remove(
+      amf_ue_context_p->tun11_ue_context_htbl,
+      (const hash_key_t) ue_context_p->amf_teid_n11);
+
+  if (INVALID_AMF_UE_NGAP_ID != amf_ue_ngap_id) {
+    h_rc = hashtable_uint64_ts_insert(
+        amf_ue_context_p->tun11_ue_context_htbl,
+        (const hash_key_t) amf_teid_n11, (uint64_t) amf_ue_ngap_id);
+  } else {
+    h_rc = HASH_TABLE_KEY_NOT_EXISTS;
+  }
+
+  if (HASH_TABLE_OK != h_rc) {
+    // TODO: this method is deprecated and will be removed once the AMF's
+    // context is migrated to map in the upcoming multi-UE PR
```

Comment:
is this still relevant? Or the comment is about moving to std c++ unordered maps?

Reply:
it was some carry forwarded from existing magma-5G, we will plan to migrate hashtable to  std c++ unordered map

Reply:
removed above comment